controller/array_action/array_mediator_svc.py

```
@@ -842,3 +871,192 @@ def get_array_fc_wwns(self, host_name):
+        return rcrelationship
+
+    def _get_rcrelationships(self, cli_volume_id, other_cli_volume_id, other_system_id, as_master):
+        endpoint_type = ENDPOINT_TYPE_AUX
+        other_endpoint_type = ENDPOINT_TYPE_MASTER
+        if as_master:
+            endpoint_type = ENDPOINT_TYPE_MASTER
+            other_endpoint_type = ENDPOINT_TYPE_AUX
+        filter_value = '{END}_vdisk_id={VDISK_ID}:' \
+                       '{OTHER_END}_vdisk_id={OTHER_VDISK_ID}:' \
+                       '{OTHER_END}_cluster_id={OTHER_CLUSTER_ID}'.format(END=endpoint_type, VDISK_ID=cli_volume_id,
+                                                                          OTHER_END=other_endpoint_type,
+                                                                          OTHER_VDISK_ID=other_cli_volume_id,
+                                                                          OTHER_CLUSTER_ID=other_system_id)
+        return self._get_lsrcrelationship(filter_value).as_list
+
+    def _get_rcrelationship(self, cli_volume_id, other_cli_volume_id, other_system_id):
+        rcrelationships = self._get_rcrelationships(cli_volume_id, other_cli_volume_id,
+                                                    other_system_id, as_master=True)
+        rcrelationships.extend(self._get_rcrelationships(cli_volume_id, other_cli_volume_id,
+                                                         other_system_id, as_master=False))
+        if len(rcrelationships) != 1:
+            logger.warning('found {0} rcrelationships for volume id {1} '
+                           'with volume id {2} of system {3}'.format(len(rcrelationships),
+                                                                     cli_volume_id,
+                                                                     other_cli_volume_id,
+                                                                     other_system_id))
+            return None
```

why only a warning?

---

should I change it to `error` and raise an exception?
in such scenario it will fail afterwards anyway

---

I'm not sure, what is the actual meaning of getting here? what is the scenario?

---

e.g. two identical rcrelationships but with opposite directions (shouldn't happen of course)

---

so its a bug it should raise an exception IMO..

---

I agree, it would be better to raise an exception here

---

will be done in CSI-3394